### PR TITLE
Invitation status logic needs to be updated#153

### PR DESCRIPTION
**Pull request by KevinFHartmann**

### CODE REVIEW

File: src/main/java/org/icgc/argo/program_service/grpc/ProgramServiceImpl.java [hunk 1]
```diff
@@ -224,6 +224,7 @@ public void listUsers(ListUserRequest request, StreamObserver<ListUserResponse>
   public void removeUser(RemoveUserRequest request, StreamObserver<RemoveUserResponse> responseObserver) {
     val programName = request.getProgramShortName().getValue();
     val email = request.getUserEmail().getValue();
+    invitationService.revoke(programName, email);
```
Comment (Buwujiu) [inline src/main/java/org/icgc/argo/program_service/grpc/ProgramServiceImpl.java]: `revoke` does not seem to be revoking the invitation.

File: src/main/java/org/icgc/argo/program_service/services/InvitationService.java [hunk 2]
```diff
@@ -45,38 +51,71 @@ public UUID inviteUser(@NotNull ProgramEntity program,
...
+    val validInvitations = invitations.stream().filter(i -> i.getStatus() != INVALID && i.getStatus() != REVOKED).
+      collect(Collectors.toList());
+    if (validInvitations.size() == 0) {
+      return Optional.empty();
+    }
+    return Optional.of(validInvitations.get(0));
+  }
+
+  public void revoke(String programShortName, String email) {
+    val previousInvitations = invitationRepository.findAllByProgramShortNameAndUserEmail(programShortName, email);
```
Comment (Buwujiu) [inline src/main/java/org/icgc/argo/program_service/services/InvitationService.java]: Why is the method called `revoke` but it's finding the invitation?

File: src/main/java/org/icgc/argo/program_service/services/InvitationService.java [hunk 3]
```diff
@@ -45,38 +51,71 @@ public UUID inviteUser(@NotNull ProgramEntity program,
...
-  public Optional<JoinProgramInviteEntity> getInvitation(String programShortName, String email) {
-    return invitationRepository.findTopByProgramShortNameAndUserEmailOrderByCreatedAtDesc(programShortName, email);
+  public Optional<JoinProgramInviteEntity> getLatestInvitation(String programShortName, String email) {
+    val invitations = listInvitations(programShortName, email);
+    val validInvitations = invitations.stream().filter(i -> i.getStatus() != INVALID && i.getStatus() != REVOKED).
+      collect(Collectors.toList());
+    if (validInvitations.size() == 0) {
+      return Optional.empty();
+    }
+    return Optional.of(validInvitations.get(0));
```
Comment (Buwujiu) [inline src/main/java/org/icgc/argo/program_service/services/InvitationService.java]: Does `get(0)` return the latest invitation?

Comment (KevinFHartmann) [reply src/main/java/org/icgc/argo/program_service/services/InvitationService.java]: In this case, yes. It gets the first element in the list, which should have been sorted in descending order by creation date.

**Pull request merged**